### ISSUE TITLE
Mismatched type for extra_field_selector option

### ISSUE THREAD

**Issue by Alexx-G** [timberio/vector#6130]
### Vector Version

```
0.11.1
```

### Vector Configuration File

```yaml
kubernetesLogsSource:
  enabled: true
  sourceId: kubernetes_logs
  rawConfig: |
    extra_field_selector = "metadata.namespace=ingress-nginx"
```

### Expected Behavior

According to documentation it's should be possible to specify multiple field selectors, possible use cases:
- include/exclude logs only from specific namespace(s);
- exclude pods with a specific name

The first use case (including only specific namespaces or excluding other namespaces) and #5846 are the only issues (detected so far) that block us from upgrading from 0.8 :tada: 

### Actual Behavior

Setting this option to array results in vector rejecting the config.
IIRC the `extra_label_selector` option has the same problem.

**Comment by MOZGIII** [comment]
It's not supposed to be an array, but rather an kubernetes-formatted filter predicates string. For example: `metadata.namespace=ingress-nginx,metadata.name=mypod`.
The reason why we don't use arrays or maps there is because the predicates string allows for more flexibility - for intance, you could use something like `metadata.namespace!=default,metadata.namespace!=other`.

This is similar to labels.

More info can be found at https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/

Imo, the best course of action here is to provide more examples in the docs: like with negating filter, or chained selectors.

**Comment by Alexx-G** [comment]
Thanks for clarifying this! So it’s just a documentation issue in this case. Currently it states that the field
is an array of strings - https://vector.dev/docs/reference/sources/kubernetes_logs/#extra_field_selector
Totally agree, a few examples and a clarification how exactly this field is being used would help a lot.

**Comment by eeyun** [comment]
closed by #6140

**Comment by eeyun** [comment]
closed by #6140
